### PR TITLE
Fix: set count status False even if check_disk_after_cmd is equal

### PR DESCRIPTION
**Pull request by harish-24**
set disk_count status to False even if check_disk_after_cmd is equal.

### CODE REVIEW

**Comment by harish-24** [comment]
**Testcase: virsh.attach_detach_disk.detach_disk.normal_test.vm_suspend**

**FAILING CASE: where check_disk_after_cmd is set not set to False** 
2016-12-16 12:26:44,312 utils_misc       L0607 DEBUG| waiting for domain virt-tests-vm1-harish to start (0.000018 secs)
2016-12-16 12:26:44,470 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:26:54,711 virt_vm          L1047 DEBUG| Login timeout expired    (output: '')
2016-12-16 12:27:03,058 virt_vm          L1047 DEBUG| Client process terminated    (status: 255,    output: 'ssh: connect to host 192.168.122.232 port 22: No route to host\n')
2016-12-16 12:27:06,694 client           L1049 DEBUG| Sending command: cat /proc/partitions
2016-12-16 12:27:06,797 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:27:06,898 libvirt          L0819 DEBUG| Find parts: ['sda', 'sda1', 'sda2', 'sda3', 'dm-0', 'dm-1']
2016-12-16 12:27:07,083 libvirt_vm       L1770 DEBUG| Destroying VM
2016-12-16 12:27:08,572 libvirt_vm       L1805 WARNI| Requested MAC address release from persistent vm virt-tests-vm1-harish. Ignoring.
2016-12-16 12:27:08,651 process          L0334 INFO | Running 'mkdir -p /var/lib/libvirt/images/harish/avocado-fvt-wrapper'
2016-12-16 12:27:08,656 process          L0435 INFO | Command 'mkdir -p /var/lib/libvirt/images/harish/avocado-fvt-wrapper' finished with 0 after 0.00086522102356s
2016-12-16 12:27:08,657 process          L0334 INFO | Running 'qemu-img create -f raw /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img 1G'
2016-12-16 12:27:08,677 process          L0421 DEBUG| [stdout] Formatting '/var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img', fmt=raw size=1073741824
2016-12-16 12:27:08,678 process          L0435 INFO | Command 'qemu-img create -f raw /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img 1G' finished with 0 after 0.0170140266418s
2016-12-16 12:27:08,678 virsh            L0645 DEBUG| Running virsh command: attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config
2016-12-16 12:27:08,679 process          L0334 INFO | Running '/bin/virsh attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config'
2016-12-16 12:27:08,724 process          L0421 DEBUG| [stdout] Disk attached successfully
2016-12-16 12:27:08,724 process          L0435 INFO | Command '/bin/virsh attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config' finished with 0 after 0.0419900417328s
2016-12-16 12:27:08,725 process          L0421 DEBUG| [stdout]
2016-12-16 12:27:08,726 virsh            L0691 DEBUG| status: 0
2016-12-16 12:27:08,726 virsh            L0692 DEBUG| stdout: Disk attached successfully
2016-12-16 12:27:08,726 virsh            L0693 DEBUG| stderr:
2016-12-16 12:27:08,752 libvirt_vm       L2023 DEBUG| Starting vm 'virt-tests-vm1-harish'
2016-12-16 12:27:09,535 utils_misc       L0607 DEBUG| waiting for domain virt-tests-vm1-harish to start (0.000017 secs)
2016-12-16 12:27:09,679 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:27:19,931 virt_vm          L1047 DEBUG| Login timeout expired    (output: '')
2016-12-16 12:27:35,839 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:27:37,244 client           L1049 DEBUG| Sending command: rpm --version
2016-12-16 12:27:37,362 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:27:37,463 client           L1049 DEBUG| Sending command: rpm -qa | grep redhat-release
2016-12-16 12:27:38,196 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:27:38,440 virsh_attach_det L0229 INFO | Suspending virt-tests-vm1-harish...
2016-12-16 12:27:38,660 virsh            L0645 DEBUG| Running virsh command: detach-disk --domain virt-tests-vm1-harish --target vdc
2016-12-16 12:27:38,661 process          L0334 INFO | Running '/bin/virsh detach-disk --domain virt-tests-vm1-harish --target vdc '
2016-12-16 12:27:43,692 process          L0421 DEBUG| [stdout] Disk detached successfully
2016-12-16 12:27:43,692 process          L0421 DEBUG| [stdout]
2016-12-16 12:27:43,692 process          L0435 INFO | Command '/bin/virsh detach-disk --domain virt-tests-vm1-harish --target vdc ' finished with 0 after 5.02776503563s
2016-12-16 12:27:43,693 virsh            L0691 DEBUG| status: 0
2016-12-16 12:27:43,693 virsh            L0692 DEBUG| stdout: Disk detached successfully
2016-12-16 12:27:43,694 virsh            L0693 DEBUG| stderr:
2016-12-16 12:27:43,754 libvirt_vm       L2113 DEBUG| Resumed VM virt-tests-vm1-harish
2016-12-16 12:27:46,824 virsh_attach_det L0042 INFO | Checking VM partittion...
2016-12-16 12:27:46,852 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:27:48,238 client           L1049 DEBUG| Sending command: cat /proc/partitions
2016-12-16 12:27:48,342 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:27:48,443 libvirt          L0819 DEBUG| Find parts: ['sda', 'sda1', 'sda2', 'sda3', 'dm-0', 'dm-1']
2016-12-16 12:27:48,444 virsh_attach_det L0051 DEBUG| Added parts: []
2016-12-16 12:27:48,615 libvirt_vm       L1770 DEBUG| Destroying VM
2016-12-16 12:27:50,214 libvirt_vm       L1805 WARNI| Requested MAC address release from persistent vm virt-tests-vm1-harish. Ignoring.
2016-12-16 12:27:53,491 virsh            L1332 DEBUG| Undefine VM virt-tests-vm1-harish
2016-12-16 12:27:53,520 virsh            L1316 DEBUG| Define VM from /var/tmp/xml_utils_temp_mQ7U3Z.xml
2016-12-16 12:27:53,567 process          L0334 INFO | Running 'rm -f /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img'
2016-12-16 12:27:53,573 process          L0435 INFO | Command 'rm -f /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img' finished with 0 after 0.000720024108887s
2016-12-16 12:27:53,799 qemu_storage     L0480 DEBUG| Checking image file /var/lib/libvirt/images/harish/avocado-fvt-wrapper/data/avocado-vt/harish/f24-ppc64le-harish.qcow2
2016-12-16 12:27:54,059 process          L0334 INFO | Running 'true'
2016-12-16 12:27:54,064 process          L0435 INFO | Command 'true' finished with 0 after 0.000538110733032s
2016-12-16 12:27:54,064 process          L0334 INFO | Running 'ps -o comm 1'
2016-12-16 12:27:54,136 process          L0421 DEBUG| [stdout] COMMAND
2016-12-16 12:27:54,137 process          L0435 INFO | Command 'ps -o comm 1' finished with 0 after 0.0685360431671s
2016-12-16 12:27:54,137 process          L0421 DEBUG| [stdout] systemd
2016-12-16 12:27:54,199 stacktrace       L0038 ERROR|
2016-12-16 12:27:54,199 stacktrace       L0041 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py:420
2016-12-16 12:27:54,200 stacktrace       L0044 ERROR| Traceback (most recent call last):
2016-12-16 12:27:54,200 stacktrace       L0044 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 206, in runTest
2016-12-16 12:27:54,200 stacktrace       L0044 ERROR|     raise exceptions.TestFail(details)
2016-12-16 12:27:54,200 stacktrace       L0044 ERROR| TestFail: See device in xml file after detach.

**Comment by harish-24** [comment]
**After fix: check_disk_after_cmd is set to False **

2016-12-16 12:54:45,531 utils_misc       L0607 DEBUG| waiting for domain virt-tests-vm1-harish to start (0.000024 secs)
2016-12-16 12:54:45,671 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:54:55,930 virt_vm          L1047 DEBUG| Login timeout expired    (output: '')
2016-12-16 12:55:06,728 client           L1049 DEBUG| Sending command: cat /proc/partitions
2016-12-16 12:55:06,832 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:55:06,933 libvirt          L0819 DEBUG| Find parts: ['sda', 'sda1', 'sda2', 'sda3', 'dm-0', 'dm-1']
2016-12-16 12:55:07,122 libvirt_vm       L1770 DEBUG| Destroying VM
2016-12-16 12:55:08,699 libvirt_vm       L1805 WARNI| Requested MAC address release from persistent vm virt-tests-vm1-harish. Ignoring.
2016-12-16 12:55:08,782 process          L0334 INFO | Running 'mkdir -p /var/lib/libvirt/images/harish/avocado-fvt-wrapper'
2016-12-16 12:55:08,788 process          L0435 INFO | Command 'mkdir -p /var/lib/libvirt/images/harish/avocado-fvt-wrapper' finished with 0 after 0.000951051712036s
2016-12-16 12:55:08,788 process          L0334 INFO | Running 'qemu-img create -f raw /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img 1G'
2016-12-16 12:55:08,810 process          L0421 DEBUG| [stdout] Formatting '/var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img', fmt=raw size=1073741824
2016-12-16 12:55:08,811 process          L0435 INFO | Command 'qemu-img create -f raw /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img 1G' finished with 0 after 0.0188341140747s
2016-12-16 12:55:08,811 virsh            L0645 DEBUG| Running virsh command: attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config
2016-12-16 12:55:08,812 process          L0334 INFO | Running '/bin/virsh attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config'
2016-12-16 12:55:08,858 process          L0421 DEBUG| [stdout] Disk attached successfully
2016-12-16 12:55:08,858 process          L0435 INFO | Command '/bin/virsh attach-disk --domain virt-tests-vm1-harish --source /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img --target vdc --driver qemu --config' finished with 0 after 0.0416421890259s
2016-12-16 12:55:08,859 process          L0421 DEBUG| [stdout]
2016-12-16 12:55:08,859 virsh            L0691 DEBUG| status: 0
2016-12-16 12:55:08,859 virsh            L0692 DEBUG| stdout: Disk attached successfully
2016-12-16 12:55:08,859 virsh            L0693 DEBUG| stderr:
2016-12-16 12:55:08,884 libvirt_vm       L2023 DEBUG| Starting vm 'virt-tests-vm1-harish'
2016-12-16 12:55:09,862 utils_misc       L0607 DEBUG| waiting for domain virt-tests-vm1-harish to start (0.000019 secs)
2016-12-16 12:55:10,019 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:55:20,263 virt_vm          L1047 DEBUG| Login timeout expired    (output: '')
2016-12-16 12:55:36,174 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:55:37,607 client           L1049 DEBUG| Sending command: rpm --version
2016-12-16 12:55:37,726 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:55:37,827 client           L1049 DEBUG| Sending command: rpm -qa | grep redhat-release
2016-12-16 12:55:38,541 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:55:38,787 virsh_attach_det L0229 INFO | Suspending virt-tests-vm1-harish...
2016-12-16 12:55:38,960 output           L0642 DEBUG| 2
2016-12-16 12:55:39,010 virsh            L0645 DEBUG| Running virsh command: detach-disk --domain virt-tests-vm1-harish --target vdc
2016-12-16 12:55:39,010 process          L0334 INFO | Running '/bin/virsh detach-disk --domain virt-tests-vm1-harish --target vdc '
2016-12-16 12:55:44,043 process          L0421 DEBUG| [stdout] Disk detached successfully
2016-12-16 12:55:44,044 virsh            L0692 DEBUG| stdout: Disk detached successfully
2016-12-16 12:55:44,044 virsh            L0693 DEBUG| stderr:
2016-12-16 12:55:44,106 libvirt_vm       L2113 DEBUG| Resumed VM virt-tests-vm1-harish
2016-12-16 12:55:47,192 virsh_attach_det L0042 INFO | Checking VM partittion...
2016-12-16 12:55:47,220 virt_vm          L1037 DEBUG| Attempting to log into 'virt-tests-vm1-harish' (timeout 240s)
2016-12-16 12:55:48,684 client           L1049 DEBUG| Sending command: cat /proc/partitions
2016-12-16 12:55:48,787 client           L1049 DEBUG| Sending command: echo $?
2016-12-16 12:55:48,890 libvirt          L0819 DEBUG| Find parts: ['sda', 'sda1', 'sda2', 'sda3', 'dm-0', 'dm-1']
2016-12-16 12:55:48,890 virsh_attach_det L0051 DEBUG| Added parts: []
2016-12-16 12:55:49,067 libvirt_vm       L1770 DEBUG| Destroying VM
2016-12-16 12:55:50,659 libvirt_vm       L1805 WARNI| Requested MAC address release from persistent vm virt-tests-vm1-harish. Ignoring.
2016-12-16 12:55:53,888 virsh            L1332 DEBUG| Undefine VM virt-tests-vm1-harish
2016-12-16 12:55:53,919 virsh            L1316 DEBUG| Define VM from /var/tmp/xml_utils_temp_ala5Fn.xml
2016-12-16 12:55:53,964 process          L0334 INFO | Running 'rm -f /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img'
2016-12-16 12:55:53,970 process          L0435 INFO | Command 'rm -f /var/lib/libvirt/images/harish/avocado-fvt-wrapper/attach.img' finished with 0 after 0.000868082046509s
2016-12-16 12:55:54,191 qemu_storage     L0480 DEBUG| Checking image file /var/lib/libvirt/images/harish/avocado-fvt-wrapper/data/avocado-vt/harish/f24-ppc64le-harish.qcow2
2016-12-16 12:55:54,563 process          L0334 INFO | Running 'true'
2016-12-16 12:55:54,568 process          L0435 INFO | Command 'true' finished with 0 after 0.000576972961426s
2016-12-16 12:55:54,569 process          L0334 INFO | Running 'ps -o comm 1'
2016-12-16 12:55:54,640 process          L0421 DEBUG| [stdout] COMMAND
2016-12-16 12:55:54,640 process          L0435 INFO | Command 'ps -o comm 1' finished with 0 after 0.0671548843384s
2016-12-16 12:55:54,641 process          L0421 DEBUG| [stdout] systemd
2016-12-16 12:55:54,709 test             L0591 INFO | PASS 2-guest_storage.attach_detach_disk.virtio.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.24.ppc64le.powerkvm-libvirt.virsh.attach_detach_disk.detach_disk.normal_test.vm_suspend

**Comment by balamuruhans** [comment]
@harish-24 I looks good, can you help in fixing,
typos:
Line 185 - servcie 
Line 222 - rhle5
Clean up:
can you use "from avocado.core import exceptions" instead of "from autotest.client.shared import error"

**Comment by harish-24** [comment]
@balamuruhans  Thanks for the suggestions. pushed with the changes

**Comment by balamuruhans** [comment]
ACK

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_disk.py [hunk 1]
```diff
@@ -305,7 +306,7 @@ def acpiphp_module_modprobe(vm, os_type):
         if disk_count_after_cmd == disk_count_before_cmd:
             check_count_after_cmd = False
     elif test_cmd == "detach-disk":
-        if disk_count_after_cmd < disk_count_before_cmd:
+        if disk_count_after_cmd <= disk_count_before_cmd:
```
Comment (will-Do) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_disk.py]: Could you give more explanation about this fix?

Comment (harish-24) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_disk.py]: @will-Do Observed that it is an actual failure. sent my mistake. will close this PR.

**Pull request closed without merging**